File: frontend/src/api/certifications.ts
// Types -------------------------------------------------------------------

export interface Domain {
  id: number
  name: string
  slug: string
  weight_pct: number
  description: string
}

export interface CertificationSummary {
  id: number
  slug: string
  display_name: string
  provider: string
  level: string
  pass_score_pct: number
}

export interface CertificationDetail extends CertificationSummary {
  prompt_context: string
  domains: Domain[]
}

// API calls ---------------------------------------------------------------

export async function listCertifications(): Promise<CertificationSummary[]> {
  const response = await fetch('/api/certifications')
  if (!response.ok) throw new Error(`Failed to fetch certifications: ${response.status}`)
  return response.json() as Promise<CertificationSummary[]>
}

export async function getCertification(slug: string): Promise<CertificationDetail> {
  const response = await fetch(`/api/certifications/${slug}`)
  if (response.status === 404) throw new Error(`Certification not found: ${slug}`)
  if (!response.ok) throw new Error(`Failed to fetch certification: ${response.status}`)
  return response.json() as Promise<CertificationDetail>
}

File: backend/app/schemas/certifications.py
from __future__ import annotations

from pydantic import BaseModel, ConfigDict


class Domain(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    slug: str
    weight_pct: int
    description: str


class CertificationSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    slug: str
    display_name: str
    provider: str
    level: str
    pass_score_pct: int


class CertificationDetail(CertificationSummary):
    prompt_context: str
    domains: list[Domain]
